File: Contests/Contest7/Q2.py
t = int(input())

for _ in range(t):
    n = int(input())
    genders = []
    aviaries = 0
    
    for i in range(n):
        action = int(input())
        
        if action == 1:
            gender = int(input())
            genders.append(gender)
        else:
            if len(genders) > 1 and len(set(genders)) == 2:
                aviaries += 1
            genders = list(set(genders))
            
    if len(genders) > 1 and len(set(genders)) == 2:
        aviaries += 1
        
    print(aviaries)

File: Contests/Contest7/Q2.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int t;
    cin >> t;
    
    while (t--) {
        int n;
        cin >> n;
        vector<int> genders;
        int aviaries = 0;
        
        for (int i = 0; i < n; i++) {
            int action;
            cin >> action;
            
            if (action == 1) {
                int gender;
                cin >> gender;
                genders.push_back(gender);
            } else {
                if (genders.size() > 1 && count(genders.begin(), genders.end(), genders[0]) != genders.size()) {
                    aviaries++;
                }
                genders.erase(unique(genders.begin(), genders.end()), genders.end());
            }
        }
        
        if (genders.size() > 1 && count(genders.begin(), genders.end(), genders[0]) != genders.size()) {
            aviaries++;
        }
        
        cout << aviaries << endl;
    }
    
    return 0;
}
